test/functional/test_framework/miniscript.py

```diff
@@ -0,0 +1,1709 @@
+                        self._j_sat, self._j_dsat,
+                        script, tag+child_x.desc)
+        return self
+
+    def construct_n(self, child_x):
+        prop_str = "u"
+        prop_str += "B" if child_x.p.B else ""
+        prop_str += "z" if child_x.p.z else ""
+        prop_str += "o" if child_x.p.o else ""
+        prop_str += "n" if child_x.p.n else ""
+        prop_str += "d" if child_x.p.d else ""
+        prop_str += "f" if child_x.p.f else ""
+        prop_str += "e" if child_x.p.e else ""
+        prop_str += "m" if child_x.p.m else ""
+        prop_str += "s" if child_x.p.s else ""
+        tag = "n" if Node._is_wrapper(child_x) else "n:"
+        self._construct(NodeType.WRAP_N, Property().from_string(prop_str),
+                        [child_x],
+                        self._n_sat, self._n_dsat,
+                        child_x._script+[OP_0NOTEQUAL], tag+child_x.desc)
+        return self
+
+    def construct_l(self, child_x):
+        prop_str = "d"
+        prop_str += "B" if child_x.p.B else ""
+        prop_str += "o" if child_x.p.z else ""
+        prop_str += "u" if child_x.p.u else ""
+        prop_str += "e" if child_x.p.f else ""
+        prop_str += "m" if child_x.p.m else ""
+        prop_str += "s" if child_x.p.s else ""
+        tag = "l" if Node._is_wrapper(child_x) else "l:"
+        script = [OP_IF, 0, OP_ELSE]+child_x._script+[OP_ENDIF]
+        self._construct(NodeType.WRAP_L, Property().from_string(prop_str),
+                        [child_x],
+                        self._l_sat, self._l_dsat,
+                        script, tag+child_x.desc)
+        return self
+
+    def construct_u(self, child_x):
+        prop_str = "d"
+        prop_str += "B" if child_x.p.B else ""
+        prop_str += "o" if child_x.p.z else ""
+        prop_str += "u" if child_x.p.u else ""
+        prop_str += "e" if child_x.p.f else ""
+        prop_str += "m" if child_x.p.m else ""
+        prop_str += "s" if child_x.p.s else ""
+        tag = "u" if Node._is_wrapper(child_x) else "u:"
+        script = [OP_IF]+child_x._script+[OP_ELSE, 0, OP_ENDIF]
+        self._construct(NodeType.WRAP_U, Property().from_string(prop_str),
+                        [child_x],
+                        self._u_sat, self._u_dsat,
+                        script, tag+child_x.desc)
+        return self
+
+    def construct_thresh(self, k, children_n):
+        n = len(children_n)
+        assert n > k > 1
+        child_is_z_count = 0
+        child_is_o_count = 0
+        child_is_s_count = 0
+        child_is_e = True
+        child_is_m = True
+        for idx, child in enumerate(children_n):
+            assert child.p.d and child.p.u
+            child_is_z_count += 1 if child.p.z else 0
+            child_is_o_count += 1 if child.p.o else 0
+            child_is_s_count += 1 if child.p.s else 0
+            child_is_e += child_is_e and child.p.e
+            child_is_m += child_is_m and child.p.m
```

Comment:
should be `=` rather than `+=` in two lines above, otherwise `child_is_e` will be converted to in tan will always be `>= 1`

Reply:
Excellent catch. Thx again.